src/main/java/com/aws/greengrass/FleetProvisioningByClaimPlugin.java

```diff
@@ -160,6 +164,12 @@ public ProvisionConfiguration updateIdentityConfiguration(ProvisionContext provi
                 writeCertificateAndKeyToPath(response, parameterMap.get(ROOT_PATH_PARAMETER_NAME).toString());
                 certificateOwnershipToken = response.certificateOwnershipToken;
             } else {
+                if (csrPrivateKeyPath == null) {
```

Comment:
catch this issue earlier by adding logic into the input validation portion of the code. Check for `Utils.isEmpty(csrPath)` just like we do here.